### PR TITLE
feat: [CDS-44978]: CD onboarding phase 2

### PR DESCRIPTION
**Pull request by PratyushGar**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/framework/utils/HelpPanelProvider.tsx [hunk 1]
```diff
@@ -11,9 +11,9 @@ import { HelpPanelContextProvider, HelpPanelEnvironment } from '@harness/help-pa
 const HelpPanelProvider: React.FC = props => {
   return (
     <HelpPanelContextProvider
-      accessToken={window.helpPanelAccessToken}
-      space={window.helpPanelSpace}
-      environment={window.helpPanelEnvironment as HelpPanelEnvironment}
+      accessToken="iMDhj0H7-zbCwnAe4avVNNFBni6-dZr4vLcaRNi5UUY"
+      space="jj62blws1bo9"
+      environment={'production' as HelpPanelEnvironment}
```
Comment (PratyushGar) [inline src/framework/utils/HelpPanelProvider.tsx:16]: added for testing, will be reverted while merge

File: src/modules/75-cd/pages/get-started-with-cd/DeployProvisioningWizard/DeployProvisioningWizard.tsx [hunk 2]
```diff
@@ -5,98 +5,316 @@
...
+  React.useEffect(() => {
+    const sectionId = (query as any).sectionId || EMPTY_STRING
+    if (sectionId?.length && WizardStepOrder.includes(sectionId)) {
+      updateStepStatusFromContextTab(sectionId)
+    } else {
+      setSelectedSectionId(DeployProvisiongWizardStepId.SelectDeploymentType)
+      updateStepStatus([DeployProvisiongWizardStepId.SelectDeploymentType], StepStatus.InProgress)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedSectionId])
```
Comment (rameshx) [inline src/modules/75-cd/pages/get-started-with-cd/DeployProvisioningWizard/DeployProvisioningWizard.tsx:138]: missing dependencies?

File: src/modules/75-cd/pages/get-started-with-cd/DeployProvisioningWizard/DeployProvisioningWizard.tsx [hunk 3]
```diff
@@ -246,91 +478,110 @@ export const DeployProvisioningWizard: React.FC<DeployProvisioningWizardProps> =
...
+      flex={{ justifyContent: 'space-between', alignItems: 'flex-start' }}
+      width="100%"
+      style={{ minHeight: '100vh', position: 'relative' }}
+    >
+      <Layout.Vertical width="100%">
+        {/* header */}
+        <Container className={css.header}>
+          <MultiStepProgressIndicator
+            progressMap={
+              new Map([
```
Comment (rameshx) [inline src/modules/75-cd/pages/get-started-with-cd/DeployProvisioningWizard/DeployProvisioningWizard.tsx:497]: use useMemo/useCallback to avoid creating this map on every render

File: src/modules/75-cd/pages/get-started-with-cd/DeployProvisioningWizard/DeployProvisioningWizard.tsx [hunk 4]
```diff
@@ -5,98 +5,316 @@
...
+    () => (
+      <Text color={Color.BLACK} padding="medium">
+        {`${getString('cd.getStartedWithCD.delegateRequiredWarning')} `}
+        <a rel="noreferrer" target="_blank" href={DOCUMENT_URL}>
+          {getString('pipeline.createPipeline.learnMore')}
+        </a>
+      </Text>
+    ),
+    [getString]
+  )
```
Comment (rameshx) [inline src/modules/75-cd/pages/get-started-with-cd/DeployProvisioningWizard/DeployProvisioningWizard.tsx:230]: `useMemo` is not required here. Also, I don't think `getString` from the strings hook is memoized.

File: src/modules/75-cd/pages/get-started-with-cd/DelegateSelectorWizard/DelegateSelectorWizard.tsx [hunk 5]
```diff
@@ -5,107 +5,147 @@
...
     }
-  })
-  const createPipelineHandler = (): void => {
-    successRefHandler?.current?.()
-    trackEvent(CDOnboardingActions.delegateInstallWizardEnd, {
-      category: Category.DELEGATE,
-      data: {
-        delegateName: delegateName.current,
-        delegateType: delegateType
+  }, [isDelegateInstalled, forwardRef])
```
Comment (rameshx) [inline src/modules/75-cd/pages/get-started-with-cd/DelegateSelectorWizard/DelegateSelectorWizard.tsx:116]: refs are not valid dependencies and this should be done using useImperativeHandle hook

File: src/modules/75-cd/pages/get-started-with-cd/DelegateSelectorWizard/CommandWithCopyField.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+          align: 'center'
+        }}
+        flex
+        className={css.verificationField}
+        width={'80%'}
+      >
+        <Text style={{ marginRight: 'var(--spacing-xlarge)', paddingLeft: '5px' }} font="small">
+          {label}
+        </Text>
+        <CopyToClipboard content={label.slice(2)} showFeedback />
```
Comment (rameshx) [inline src/modules/75-cd/pages/get-started-with-cd/DelegateSelectorWizard/CommandWithCopyField.tsx:27]: Why are first two chars sliced? Should this logic be present in the parent component?

File: src/modules/75-cd/pages/get-started-with-cd/CreateKubernetesDelegateWizard/StepProcessing.tsx [hunk 7]
```diff
@@ -121,13 +121,24 @@ const StepProcessing: FC<StepDelegateData> = props => {
...
-                onClick={() => setTroubleShootVisible(!isTroubleShootVisible)}
-                color={Color.BLUE_700}
-              >
-                {getString('delegates.delegateNotInstalled.tabs.commonProblems.troubleshoot')}
-              </Text>
+              <Layout.Horizontal className={css.textPadding}>
+                <Button
+                  variation={ButtonVariation.SECONDARY}
+                  onClick={() => {
+                    verifyHeartBeat()
```
Comment (rameshx) [inline src/modules/75-cd/pages/get-started-with-cd/CreateKubernetesDelegateWizard/StepProcessing.tsx:128]:
```suggestion
                    verifyHeartbeat()
```

File: src/modules/75-cd/pages/get-started-with-cd/ConfigureService/ManifestRepoTypes/SelectRepository.tsx [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+          addClearBtn={true}
+          onQueryChange={setQuery}
+          placeholder={fetchingRepositories ? getString('cd.fetchingRepository') : getString('cd.selectRepository')}
+        />
+        {showValidationErrorForRepositoryNotSelected ? (
+          <Container padding={{ top: 'xsmall' }}>
+            <FormError
+              name={'repository'}
+              errorMessage={getString('common.getStarted.plsChoose', {
+                field: `a ${getString('repository').toLowerCase()}`
```
Comment (rameshx) [inline src/modules/75-cd/pages/get-started-with-cd/ConfigureService/ManifestRepoTypes/SelectRepository.tsx:131]: entire string should be translated

File: src/modules/75-cd/pages/get-started-with-cd/ConfigureService/ManifestRepoTypes/SelectRepository.tsx [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+          orgIdentifier,
+          connectorRef: `${ACCOUNT_SCOPE_PREFIX}${validatedConnectorRef}`
+        }
+      })
+    }
+  }, [accountId, fetchRepositories, orgIdentifier, projectIdentifier, validatedConnectorRef])
+
+  useEffect(() => {
+    setRepositories(repoData?.data)
+  }, [repoData?.data])
```
Comment (rameshx) [inline src/modules/75-cd/pages/get-started-with-cd/ConfigureService/ManifestRepoTypes/SelectRepository.tsx:72]: `repositories` can be derived from `repoData?.data`, no need for additional useState, useEffect

File: src/modules/75-cd/pages/get-started-with-cd/ConfigureService/ManifestRepoTypes/SelectGitProvider.tsx [hunk 10]
```diff
@@ -119,32 +120,29 @@ const SelectGitProviderRef = (
...
-  }, [gitProvider?.type])
 
-  const markOAuthAsFailed = useCallback(() => {
-    /*istanbul ignore next */
-    setOAuthStatus(Status.FAILURE)
-    clear()
-    showError(getString('connectors.oAuth.failed'))
-  }, [])
+  useEffect(() => {
+    setGitProvider(selectedGitProvider)
```
Comment (rameshx) [inline src/modules/75-cd/pages/get-started-with-cd/ConfigureService/ManifestRepoTypes/SelectGitProvider.tsx:141]: can `selectedGitProvider` be used directly? this effect and `setGitProvider` state may not be required

File: src/modules/75-cd/pages/get-started-with-cd/ConfigureService/ManifestRepoTypes/RightDrawer/RightDrawer.tsx [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+    (e?: SyntheticEvent<HTMLElement, Event> | undefined): void => {
+      e?.persist()
+      setIsOpen(false)
+    },
+    [setIsOpen]
+  )
+
+  const onCloseDrawer = React.useCallback(() => {
+    closeDrawer()
+  }, [closeDrawer])
```
Comment (rameshx) [inline src/modules/75-cd/pages/get-started-with-cd/ConfigureService/ManifestRepoTypes/RightDrawer/RightDrawer.tsx:29]: Why is this needed?

File: src/modules/75-cd/pages/get-started-with-cd/ConfigureService/ConfigureService.tsx [hunk 12]
```diff
@@ -0,0 +1,642 @@
...
+        forwardRef.current = {
+          submitForm: formikRef?.current?.submitForm
+        }
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [
+    formikRef?.current?.values,
+    formikRef?.current?.setFieldTouched,
+    selectGitProviderRef?.current?.values,
```
Comment (rameshx) [inline src/modules/75-cd/pages/get-started-with-cd/ConfigureService/ConfigureService.tsx:307]: refs are not valid dependencies

**Review by rameshx — CHANGES_REQUESTED** [review]
